apps/spawn/priv/protos/eigr/functions/protocol/actors/actor.proto

```diff
@@ -43,10 +43,39 @@ message ActorState {
     google.protobuf.Any state = 2;
 }
 
-message Actor {
-    string name = 1;
+message ActorSettings {
+
+    // Indicates if actor´s is abstract or non abstract.
+    bool abstract = 1;
+
+    // Indicates whether an actor's state should be persisted in a definitive store.
     bool persistent = 2;
-    ActorState state = 3;
-    ActorSnapshotStrategy snapshot_strategy = 4;
-    ActorDeactivateStrategy deactivate_strategy = 5;
+
+    // Snapshot strategy
+    ActorSnapshotStrategy snapshot_strategy = 3;
+
+    // Deactivate strategy
+    ActorDeactivateStrategy deactivate_strategy = 4;
+}
+
+message ActorId {
+    // The name of a Actor Entity.
+    string name = 1;
+
+    // Name of a ActorSystem
+    string system = 2;
+
+    // Actor identifier in persistent storage. For internal proxy use only
+    string hash_key = 3;
```

Comment:
perhaps we can offset internal fields? with evolution for public field we would not mix them?

Reply:
> perhaps we can offset internal fields? with evolution for public field we would not mix them?

I agree. We can remove it here

Reply:
I meant, offset with the proto field numbers; not removing it. I think its fine to have fields if needed internally. Alas, I'm not sure if we could even have an proxy-internal version without the "private" field, with not colliding proto field by their id.

Reply:
I decided to remove it while I study this further. We can include if necessary in another PR.